File: Samples/State Pattern/Sample_1 - CS/CalmState.cs
namespace Sample
{
    internal class CalmState : State
    {
        public override void Enter()
        {
            Console.WriteLine($"[{_enemy.GetName}.CalmState]: called Enter!");
        }

        public override void Exit()
        {
            Console.WriteLine($"[{_enemy.GetName}.CalmState]: called Exit!");
        }

        public override void Update()
        {
            Console.WriteLine($"[{_enemy.GetName}.CalmState]: called Update!");
        }
    }
}

File: Samples/State Pattern/Sample_1 - CPP/CalmState.cpp
#include "CalmState.h"

void CalmState::Enter()
{
	cout << "[" << m_enemy->GetName() << ".CalmState] called Enter!" << endl;
}

void CalmState::Update()
{
	cout << "[" << m_enemy->GetName() << ".CalmState] called Update!" << endl;
}

void CalmState::Exit()
{
	cout << "[" << m_enemy->GetName() << ".CalmState] called Exit!" << endl;
}
